.github/workflows/unit-tests.yml

```diff
@@ -22,18 +22,38 @@ jobs:
 
     steps:
 
-      - uses: actions/setup-python@v2
+      - name: Setup Python
+        uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python-version }}
-      - run: python -m pip install --upgrade pip
 
-      - uses: actions/checkout@v2
+      - name: Upgrade pip
+        run: python -m pip install --upgrade pip
 
-      - run: python -m pip install .[test]
+      - name: Checkout code
+        uses: actions/checkout@v2
 
-      - run: python -m pytest tests --cov=endaq
+      - name: Get pip Cache Location
+        id: pip-cache
+        run: echo "::set-output name=dir::$(pip cache dir)"
 
-      - uses: codecov/codecov-action@v2
+      - name: Load pip Cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.pip-cache.outputs.dir }}
+          key: ${{ runner.os }}-Python${{ matrix.python-version }}-pip-${{ hashFiles('**/setup.py', './.github/workflows/unit-tests.yml') }}
```

Comment:
If we're going to cache, I'd like the date to be included in the key.  Otherwise, I'd be concerned that our dependencies might just get more and more out of date, to the point we gain nothing from caching.

Reply:
So one detail I want to make clear (Connor I know you know this; this is for others): I *did* change the install to do `pip install --upgrade --upgrade-strategy=eager ...`, so if the cache has old dependencies, pip will just fetch the new ones and **not use** the cache. So there's no risk there.

Re: adding in the date, I don't see a portable (i.e., supported on ubuntu, macOS and windows) datetime action in the GitHub Actions market, so tbh I don't know what the best way would be to add that in... :\ 

But taking the bright side of what you're saying, the worst thing that happens is that *nothing changes*. And the best thing that happens is that we get speedups. Because of this, I'd rather push it through now as-is and fix it later when we find a good solution.

Reply:
I'll mark this as resolved, and make an issue for this.

Reply:
actually my bad; you can just run a bash command (or powershell, apparently?) and it'll run fine on any platform.

-> addressed in 5c0f430506c4e668b693889edb3b1b2e87f8e3d1

Reply:
well... kind of. daily seems too frequent, to the point that the cache may not ever get used. I'd prefer weekly, but I don't see an easy way to do that. @CFlaniganMide how do you feel about monthly resets?

Reply:
monthly is probably fine.  I feel like a oneline python call could do it.

`python -c "import datetime; now = datetime.datetime.now(); print(f'{now.year}-{now.month}-{now.day % 7}')"`

not exactly weekly, but pretty close.

Reply:
put in an inline comment